index.js

```diff
@@ -3983,6 +3983,26 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Returns the substrings of its second argument separated by evaluated
+  //. matches of its first argument.
+  //.
+  //. See also [`splitOn`](#splitOn).
+  //.
+  //. ```javascript
+  //. > S.splitOnRegex(/\s+/, 'foo  bar baz   qux')
+  //. ['foo', 'bar', 'baz', 'qux']
+  //. ```
+  function splitOnRegex(pattern, s) {
+    return s.split(pattern);
+  }
+  S.splitOnRegex =
+  def('splitOnRegex',
+    {},
+    [$.RegExp, $.String, $.Array($.String)],
+    splitOnRegex);
```

Comment:
Ah I see, I'm fine with that too. Out of curiosity, is there a reason behind why there are two separate functions? Are we trying to avoid polymorphic parameters? 

Thanks again for all the feedback.